mocacares_backend/event_platform/views/views_event.py

```diff
@@ -72,7 +78,10 @@ def get_events(request):
     if event_type is not None:
         events = events.filter(event_type=event_type)
     if search_key is not None:
-        events = events.filter(Q(title__icontains=search_key) | Q(description__icontains=search_key))
+        keyword_matching_events = events.filter(Q(title__icontains=search_key) | Q(description__icontains=search_key))
+        keyword_matching_users = User.objects.filter(Q(username__icontains=search_key) | Q(statement__icontains=search_key))
+        events_posted_by_matching_users = events.filter(poster__in=keyword_matching_users)
+        events = keyword_matching_events | events_posted_by_matching_users
```

Comment:
is this operator equivalent to union function?

Reply:
yes